client/core/core_test.go

```diff
@@ -4361,6 +4374,54 @@ func TestHandleTradeSuspensionMsg(t *testing.T) {
 	}
 }
 
+func orderNoteFeed(tCore *Core, t *testing.T) chan *OrderNote {
+	t.Helper()
+	ch := tCore.NotificationFeed()
+	outChan := make(chan *OrderNote)
+	go func() {
+		for start := time.Now(); time.Since(start) < time.Second; {
+			select {
+			case n := <-ch:
+				ordNote, ok := n.(*OrderNote)
+				if ok {
+					outChan <- ordNote
+					return
+				}
+			case <-tCtx.Done():
+				return
+			case <-time.NewTimer(time.Second).C:
+				t.Fatal("timed out waiting for notifications")
+				return
+			}
+		}
+	}()
+	return outChan
+}
+
+func verifyRevokeNotification(ch chan *OrderNote, expectedOrderNote *OrderNote, t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case actualOrderNote := <-ch:
+				if expectedOrderNote.SubjectText != actualOrderNote.SubjectText {
+					t.Fatalf("SubjectText mismatch. %s != %s", actualOrderNote.SubjectText,
+						expectedOrderNote.SubjectText)
+				}
+				return
+			case <-tCtx.Done():
+				return
+			case <-time.NewTimer(time.Second).C:
+				t.Fatal("timed out waiting for OrderNote notification")
+				return
+			}
+		}
```

Comment:
If I'm not mistaken, the `WaitGroup`, the goroutine, and the `for` loop are all unnecessary and identical to just doing the select block alone.

Reply:
The problem I encountered without a WaitGroup was that it seemed the test could complete before timing out in the case that the notification was not sent. I ensured that omitting the notification cause a test failure from timing out. Without the wait the test would intermittently pass or fail.

Reply:
I don't follow either.  How could there be more than one iteration of the `for` loop?  Each case has a return and the timeout will fire if the other cases do not?

Also, Fatal/Fatalf are prohibited in goroutines that are not the main test goroutine.

```go
func verifyRevokeNotification(ch chan *OrderNote, expectedSubject string, t *testing.T) {
	select {
	case actualOrderNote := <-ch:
		if expectedSubject != actualOrderNote.SubjectText {
			t.Fatalf("SubjectText mismatch. %s != %s", actualOrderNote.SubjectText,
				expectedSubject)
		}
		return
	case <-tCtx.Done():
		return
	case <-time.After(time.Second):
		t.Fatalf("timed out waiting for OrderNote notification")
		return
	}
}
```

Isn't the above correct?

Reply:
That does work. I thought I had tried that as I went through many different iterations, but I should've tested buck's assertion.